docs/development/packaging.md

```diff
@@ -1,27 +1,115 @@
+tools a regular Ralph developer already has. This required introducing a few
+compromises:
 
-Note: Internally we run `dch` debian tool to help us generate build number, and generate debian/changelog. file.
-After issuing a new 'major' release it is required to manually pre-set new version in the `debian/changelog` 
-file using dch utility. Remember to commit this file after issuing new bulid!
+* Everything is done in docker containers to avoid forcing developers to
+  install Ubuntu
+* Since IO performance for docker mounted volumes experiences severe problems
+  on certain platforms, containers are built every time a package maintenance
+  operation is performed.
+* Official Ralph repository is owned by [Allegro], so publishing a package
+  requires sharing Allegro's credentials for Bintray which according to their
+  policy cannot be done. Therefore the script for that is going to only be
+  present in the CI system that is available to maintainers inside Allegro.
 
-Note 2: Probably we will fix this somehow in the future to be fully automated ;-)
 
-### Sources
+## Building packages
 
-* `/packaging` contains buildscripts for building and uploading to bintray.com server
-* `/debian` is here because it requires some structure from source code being in place.
-* `debian/rules` and `debian/control` has some build configuration
-* `debian/changelog` - tracks build number
+Building Ralph package can be done in two modes: snapshot and release. Those
+modes produce packages for testing and for production uses respectively. The
+Makefile available in Ralph provides a target for each of the modes.
 
+
+### Building snapshot packages
+
+Snapshot packages can be built for any commit and are meant to be used for
+either developers or a CI to test some code or configuration changes. In order
+to build a snapshot package run:
+
+```bash
+make build-snapshot-package
+```
+
+The built package will be put into the `build` catalog and will have the name
+that will look like `ralph-core_<DATE>.<PATCH>-<BANCH>-SNAPSHOT_amd64.deb`.
+
+
+### Building release packages
+
+Release packages are meant to be used on production therefore they should be
+build from the tagged HEAD of the `ng` branch. This, however, is not strictly
+controlled by the tool chain to handle different edge cases.
+
+In order to build a release package run:
+
+```bash
+make build-package
+```
+
+ > If a release package is meant to be published, a maintainer should release
+ > a new version prior to building one. The way to do that is described bellow.
+
+
+## Releasing and publishing new versions
+
+Properly releasing versions is one of the key duties of every Ralph maintainer.
+
+Like the rest of the package management operations, releasing and publishing a
+new version is automated and does not require specific environment to be
+performed.
+
+However, it is necessary for a Ralph maintainer to have their development tools
```

Comment:
... and remove this paragraph(merged with the first paragraph)

Reply:
Done